src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -298,5 +298,71 @@ public void printInfo(List<Person> personList) {
 
     }
 
+    public void checkByGenderMap() {
+        List<String> male = new ArrayList<>();
+        List<String> female = new ArrayList<>();
+
+        for (Person person : persons) {
+            if (person.getGender() == Gender.MALE) {
+                male.add(person.getName());
+            } else {
+                female.add(person.getName());
+            }
+        }
+
+        Map<Gender, List<String>> genderListMapMale = new HashMap<>();
+        Map<Gender, List<String>> genderListMapFemale = new HashMap<>();
+
+        genderListMapMale.put(Gender.MALE, male);
+        genderListMapFemale.put(Gender.FEMALE, female);
+
+        if (female.size() > 0 && male.size() > 0) {
+            System.out.println(genderListMapFemale + " " + genderListMapMale);
+        } else if (female.size() != 0) {
+            System.out.println(genderListMapFemale);
+        } else if (male.size() != 0) {
+            System.out.println(genderListMapMale);
+        } else {
+            System.out.println("Statistics is empty");
+        }
+
+    }
+
+    public void checkMapByPersonProperty(PersonProperty personProperty) {
+        Map<String, List<Person>> stringListMap = new HashMap<>();
+
+        for (Person person : persons) {
+            String key = generateKey(personProperty, person);
+
+            List<Person> personList = new ArrayList<>();
+            personList.add(person);
+
+            List<Person> personListGetKey = stringListMap.get(key);
```

Comment:
This comment related to the logic from lines 340 till 348.
This logic will work fine, but from code perspective it could be done better.
The reason why it is not good because in the line 340 you are getting value from map using key and then in the line 342 you checking this key again. You are checking the key two times, but this can be done with one check.
My suggestions:
1. (simple way) rewrite this logic (from lines 340 till 348) using hashmap getOrDefault() method.
2. (best way) rewrite this logic (from lines 340 till 348) using hashmap compute() method.